test/libsolidity/semanticTests/abiEncoderV2/dirty_memory_dynamic_array_and_calldata_string.sol

```diff
@@ -0,0 +1,34 @@
+pragma abicoder v2;
+
+contract C {
+    struct Data {
+        uint256[] a;
+        string b;
+    }
+
+    function f(Data calldata data, uint256[] calldata a) public pure returns (bytes memory) {
+        // Make memory dirty (numbers come from trial & err)
+        for (uint pos = 0x80; pos < 0x880; pos += 0x20)
```

Comment:
I think it should be enough to clear between `mload(0x40)` (i.e. the free mem ptr) and `add(mload(0x40), 0x800)`.

That would be more future-proof, and would not break if we start reserving more memory at the beginning. I guess it could break even today if you happen to have some vars that need moving to memory.

Reply:
I couldn't go lower than 0x5E until the tests started to respond with "FAILURE"

Reply:
Ooh, I completely misunderstood you :laughing: